### PR TITLE
class metadata

### PR DESCRIPTION
**Pull request by DavidBadura**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Has tests?    | yes
| Fixed tickets | #223, #140

* [x] add class metadata
* [x] add annotation metadata driver
* [x] allow different metadata drivers
* [x] add `ignore field` annotation
* [x] remove `audited_entities` configuration
* [x] remove `global_ignore_properties`
* [x] remove hardcoded default entity manager
* [x] refactor the tests so that they are deterministic
* [x] add build symfony container test
* [x] update documentation

### CODE REVIEW

File: tests/SimpleThings/Tests/EntityAudit/Fixtures/Core/PetAudit.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace SimpleThings\EntityAudit\Tests\Fixtures\Core;
 
 use Doctrine\ORM\Mapping as ORM;
+use SimpleThings\EntityAudit\Mapping\Annotation\Auditable;
```
Comment (smoench) [inline tests/SimpleThings/Tests/EntityAudit/Fixtures/Core/PetAudit.php]: Unused

**Comment by Soullivaneuh** [comment]
@DavidBadura So from now we can only use annotation? How about entities defined from configuration?

Example: http://symfony.com/doc/current/doctrine.html#add-mapping-information (Yaml and XML examples).

**Comment by DavidBadura** [comment]
The metadata from entity audit has nothing to do with doctrine`s metadata. You can use yaml and xml to configure your entities, but (only) for auditing you need to use annotation for now.

I have created an issue for the yaml driver: https://github.com/simplethings/EntityAuditBundle/issues/262

**Comment by bendavies** [comment]
@DavidBadura the listeners are no longer automatically registered after this change, making the bundle integration useless. did you realise?

**Comment by DavidBadura** [comment]
@bendavies Yes, I know it. I don't have the time to finish it. We have already an issue for this problem https://github.com/simplethings/EntityAuditBundle/issues/268

**Comment by bendavies** [comment]
alright, thanks!

**Comment by mikeyudin** [comment]
First: Awesome work with this feature, but 1 question:  Why was global_ignore_properties removed?  This seems like a nice feature to have so devs don't have to add the ignore annotation to a ton of entities.

**Pull request merged**